### PR TITLE
fix: Command/Combobox TypeError and Unclickable/Disabled items + Documentation

### LINKED ISSUE #2944: [Bug]: Adding command
### Describe the bug

When utilizing a Command component that has been added/updated as of 03/07/2024 ~6PM PST,
This is due to cmdk making a new release with breaking changes: https://github.com/pacocoursey/cmdk/releases

This can be fixed by pinning cmdk version to 0.2.1. Haven't investigated shadcn fix yet.

### Affected component/components

Command

### How to reproduce

1. Run `npx shadcn-ui@latest add command` after 03/07/2024
2. Try to use command component

### Codesandbox/StackBlitz link

https://codesandbox.io/p/devbox/shadcn-playground-forked-4xxqcw?workspaceId=2d1a1544-9936-4d36-a113-0092357e5e51

### Logs

_No response_

### System Info

```bash
cmdk v1.0.0
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

### PR DESCRIPTION
**Pull request by jhnguyen521**
Fixes #2944 due to latest cmdk update with breaking changes for the Command component. Until this fix is merged, all new command components that install cmdk@1.00 will be broken and items will not be selectable. People who also have existing components and update to cmdk 1.00 and don’t have a `<CommandList>` will experience crashes upon opening up the component. See issue above for more details.

- Additionally handling the data-disabled property better in all components, should be backwards compatible
^ Happy to not do this for all other components, but I figure it's better to just do this preventatively if it's backwards compatible. Either way, things will break for data-disabled="false" for all existing `data-[disabled]`
- Applied same fix for aria-selected
- Fixing broken examples for command component

### CODE REVIEW

**Comment by kevinmitch14** [comment]
This PR handles the proper styling via data attributes + dependancy bump.

https://github.com/shadcn-ui/ui/pull/2626

**Comment by jhnguyen521** [comment]
Fixed crlf line endings to lf. May be worth documenting this in contributing, but pnpm seems to output files with \r\n instead of just \r on windows machines, ignoring tsconfig compiler options

**Comment by kevinmitch14** [comment]
The component specific issues have already been fixed in https://github.com/shadcn-ui/ui/pull/2626 🤔

**Comment by jhnguyen521** [comment]
> The component specific issues have already been fixed in #2626 🤔

After you added the fix from this and the other issue afterwards in a later commit, then yes :) Also, aria is already set in cmdk and we can be relatively certain that will always be there for accessibility reasons, not sure how opinionated we should be about using their aria/data attribute. Personally, I think it's fine to leave as is but what do I know.

Also, this PR updates necessary documentation to handle other breaking changes in cmdk.

**Comment by jhnguyen521** [comment]
Happy to do a version bump in this PR as well.

**Comment by kevinmitch14** [comment]
I am not too sure why you are hinting that I have "copied" your work or something?

With all due respect, the original #2626 PR was opened in January as a follow up from #1980 that was opened in November 2023. I have been monitoring this via an issue opened in the `cmdk` repo itself.

After my initial comment on this PR, I noticed that you made some extra commits, specifically, extra changes that were made in #2626 that had not been made in this PR. I see that these commits have now been wiped out as you cleaned the commit history...

You also commented on my PR which you have since deleted...I don't understand

**Comment by jhnguyen521** [comment]
> After my initial comment on this PR, I noticed that you made some extra commits, specifically, extra changes that were made in https://github.com/shadcn-ui/ui/pull/2626 that had not been made in this PR. I see that these commits have now been wiped out as you cleaned the commit history...

Yes, the change I made were applying my bug fix to the aria-selected attributes, which is the purpose of this PR? Whereas your PR was originally just changing aria-selected -> data-selected.
Huh? Cleaned the commit history? Probably because just did a rebase after some new stuff was merged in, but all commits should be as they have...

> You also commented on my PR which you have since deleted...I don't understand
My comment was on being opinionated about aria-selected vs data-selected, but I realized that I was incorrect about some assumptions I was saying that made my comment irrelevant so I just deleted it :)

**Review by fudy-pablo — APPROVED** [review]

File: apps/www/registry/default/ui/select.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ const SelectItem = React.forwardRef<
   <SelectPrimitive.Item
     ref={ref}
     className={cn(
-      "relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
+      "relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled='true']:pointer-events-none data-[disabled='true']:opacity-50",
```
Comment (CentraXx) [inline apps/www/registry/default/ui/select.tsx:121]: This does not work properly. Please test this...

File: apps/www/registry/default/ui/dropdown-menu.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ const DropdownMenuItem = React.forwardRef<
   <DropdownMenuPrimitive.Item
     ref={ref}
     className={cn(
-      "relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
+      "relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled='true']:pointer-events-none data-[disabled='true']:opacity-50",
```
Comment (CentraXx) [inline apps/www/registry/default/ui/dropdown-menu.tsx:86]: This does not work properly. Please test this...

**Review by CentraXx — COMMENTED** [review]
For many components which are not from cmdk the applied changes do not work and cause the disable state to not work anymore.
Means the fix: data-[disabled='true'] does not work for all components.

**Comment by jhnguyen521** [comment]
> For many components which are not from cmdk the applied changes do not work and cause the disable state to not work anymore. Means the fix: data-[disabled='true'] does not work for all components.

Good catch, thanks! I'll probably just revert changes to all other components since I'd probably have to make a custom class in css to get this to work for all while still being in the classNames.

**Comment by benjamin-guibert** [comment]
Thank you for the fix!

**Comment by digitaltim-de** [comment]
Same Issue on the Example Filter Function, that dont works anymore too.

**Comment by nopitown** [comment]
Thanks for the fix!

**Comment by logan-qiu** [comment]
@jhnguyen521 thank you! I am pretty new to shadcn and figured it will lead to a crash if I follow the official example in `ui/apps/www/app/(app)/examples/forms/account/account-form.tsx` for the language selector.
I had to wrap CommandGroup with CommandList to avoid the crash. However, none of the selection was selectable.

Hopefully this fix will be merged into the main branch soon! kudos to you.

**Comment by valascus** [comment]
When will this be merged?
Thank you

**Review by wootra — APPROVED** [review]

**Review by wootra — APPROVED** [review]
reviewed the code and tested in the code space. 
looks good to me.

**Comment by kaiguang** [comment]
Copied the code from the official website and found out it's not working with `cmdk@1.0.0`, maybe we can lock the package version with the install instruction?

Screenshot 2024-05-27 at 21 04 41

**Comment by shadcn** [comment]
@jhnguyen521 Any chance you can fix the conflicts here please? Let's merge.

**Comment by kevinmitch14** [comment]
@shadcn check out #2626 which is a follow up of #1980. Both pending for a long time addressing this issue.

**Comment by jhnguyen521** [comment]
> @jhnguyen521 Any chance you can fix the conflicts here please? Let's merge.

Conflicts are fixed :)

**Review by ivan2214 — APPROVED** [review]
LGTM

File: apps/www/registry/default/ui/command.tsx [hunk 3]
```diff
@@ -117,7 +117,7 @@ const CommandItem = React.forwardRef<
   <CommandPrimitive.Item
     ref={ref}
     className={cn(
-      "relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-accent data-[selected=true]:text-accent-foreground data-[disabled=true]:opacity-50",
+      "relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none aria-[selected='true']:bg-accent aria-[selected='true']:text-accent-foreground data-[disabled='true']:pointer-events-none data-[disabled='true']:opacity-50",
```
Comment (shadcn) [inline apps/www/registry/default/ui/command.tsx:120]: Can we use the new data-selected attribute instead if aria? Thank you.

**Comment by kotAPI** [comment]
Any updates on this?

**Comment by saver711** [comment]
WE NEED THIS !!!

**Comment by shadcn** [comment]
@kevinmitch14 I merged https://github.com/shadcn-ui/ui/pull/2626. Anything else we're missing? \cc @jhnguyen521

**Comment by kevinmitch14** [comment]
@shadcn it looks like some changes made in #2626 were reverted in https://github.com/shadcn-ui/ui/pull/4181/ by accident.

**Comment by kevinmitch14** [comment]
@shadcn it looks like some changes made in #2626 were reverted in https://github.com/shadcn-ui/ui/pull/4181/ by accident. (`command.tsx`, `model-selector.tsx`, `cmdk` package was reverted back to v0.2.0 files)

**Comment by shadcn** [comment]
@kevinmitch14 Thanks. On it.

**Review by shadcn — APPROVED** [review]

**Pull request merged**